src/components/DepositWidget/useRowActions.tsx

```diff
@@ -1,30 +1,104 @@
-import { SetStateAction, Dispatch } from 'react'
+import { SetStateAction, Dispatch, useReducer } from 'react'
 import { toast } from 'react-toastify'
 import BN from 'bn.js'
 
 import { depositApi, erc20Api } from 'api'
 import { Mutation, TokenBalanceDetails } from 'types'
-import { HIGHLIGHT_TIME, ALLOWANCE_MAX_VALUE, ZERO } from 'const'
+import { ALLOWANCE_MAX_VALUE, ZERO } from 'const'
 import { useWalletConnection } from 'hooks/useWalletConnection'
+
 import { formatAmount, formatAmountFull, log, getToken, assert, safeFilledToken } from 'utils'
 import { txOptionalParams } from 'utils/transaction'
 
 const ON_ERROR_MESSAGE = 'No logged in user found. Please check wallet connectivity status and try again.'
 
+/************************************************************** */
+// Reducer specific typings
+
+export interface TokenLocalState {
+  enabling: Set<string>
+  highlighted: Set<string>
+  claiming: Set<string>
+}
+
+const enum ActionTypes {
+  SET_ENABLING = 'enabling',
+  SET_CLAIMING = 'claiming',
+  SET_HIGHLIGHTED = 'highlighted',
+  SET_HIGHLIGHTED_AND_CLAIMING = 'highlighted_and_claiming',
+}
+
+interface Actions {
+  type: ActionTypes
```

Comment:
I think this approach is totally fine, although I'm more in favour of  declaring how we do in other parts of the app :
`type: 'enabling' | 'claiming' | ...`  instead of an enum cause it's less verbose and typescript does the checks also, and even the the intellisense/suggestions work better